app/components/autocomplete/emoji_suggestion/emoji_suggestion.tsx

```diff
@@ -0,0 +1,235 @@
+    updateValue,
+    onShowingChange,
+    rootId,
+    value,
+    nestedScrollEnabled,
+    skinTone,
+}: Props) => {
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+    const serverUrl = useServerUrl();
+    const flatListStyle = useMemo(() =>
+        [style.listView, {maxHeight: maxListHeight}]
+    , [style, maxListHeight]);
+
+    const emojis = useMemo(() => {
+        const emoticons = new Set<string>();
+        for (const [key, index] of EmojiIndicesByAlias.entries()) {
+            const skin = getSkin(Emojis[index]);
+            if (!skin || skin === skinTone) {
+                emoticons.add(key);
+            }
+        }
+
+        for (const custom of customEmojis) {
+            emoticons.add(custom.name);
+        }
+
+        return Array.from(emoticons);
+    }, [skinTone, customEmojis]);
+
+    const searchTerm = useMemo(() => {
+        const match = value.substring(0, cursorPosition).match(EMOJI_REGEX);
+        return match?.[3] || '';
+    }, [value, cursorPosition]);
+
+    const fuse = useMemo(() => {
+        return new Fuse(emojis, FUSE_OPTIONS);
+    }, [emojis]);
+
+    const data = useMemo(() => {
+        const searchTermLowerCase = searchTerm.toLowerCase();
+
+        if (searchTerm.length < 3) {
+            return [];
+        }
+
+        const sorter = (a: string, b: string) => {
+            return compareEmojis(a, b, searchTermLowerCase);
+        };
+
+        const fuzz = fuse.search(searchTermLowerCase);
+
+        if (fuzz) {
+            const results = fuzz.reduce((values, r) => {
+                const score = r?.score === undefined ? 1 : r.score;
+                const v = r?.matches?.[0]?.value;
+                if (score < 0.2 && v) {
+                    values.push(v);
+                }
+
+                return values;
+            }, [] as string[]);
+
+            return results.sort(sorter);
+        }
+
+        return [];
+    }, [fuse, searchTerm]);
+
+    const showingElements = Boolean(data.length);
+
+    const completeSuggestion = useCallback((emoji: string) => {
+        const emojiPart = value.substring(0, cursorPosition);
+
+        if (emojiPart.startsWith('+:')) {
+            handleReactionToLatestPost(serverUrl, emoji, true, rootId);
+            updateValue('');
+            return;
+        }
+        if (emojiPart.startsWith('-:')) {
+            handleReactionToLatestPost(serverUrl, emoji, false, rootId);
+            updateValue('');
+            return;
+        }
+
+        // We are going to set a double : on iOS to prevent the auto correct from taking over and replacing it
+        // with the wrong value, this is a hack but I could not found another way to solve it
+        let completedDraft: string;
+        let prefix = ':';
+        if (Platform.OS === 'ios') {
+            prefix = '::';
+        }
+
+        const emojiData = getEmojiByName(emoji);
+        if (emojiData?.image && emojiData.category !== 'custom') {
```

Comment:
hmm, might be nothing, but where does the property 'category' come from?  should it be optionally chained?

Reply:
If we get the `.image` part means that the emoji comes from the predefined ones, and therefore it has a category. In either case, if the category is undefined, it should still work correctly.